Automattic/wp-calypso
# Stepper: Framework tracking for remaining step-navigation controls

Related to https://github.com/Automattic/wp-calypso/issues/91755
Part of addressing https://github.com/Automattic/wp-calypso/issues/94108

## Proposed Changes

* Refactors `recordSubmitStep` into `recordStepNavigation` 
* Adds bindings for the remaining events/controls (`exitFlow`, `goBack`, `goNext`, `goToStep`) in `useStepNavigationWithTracking`
* Unit tests for `useStepNavigationWithTracking` confirming `recordStepNavigation` and the wrapped callbacks are called with correct arguments

## Why are these changes being made?


* Stepper to have a default Tracks event logging for all the navigation controls.

## Testing Instructions



* Ensure that navigating steps in Stepper records all the respective events with the correct properties
  * test `/setup/onboarding`, `/setup/newsletter`
    * double-check the `submit` event carefully as the callback has been abstracted
  * [connect-domain](https://github.com/Automattic/wp-calypso/pull/93275) flow extends the `submit` event props with `domain` and `provider`. Ensure these are logged correctly
  * Use the `/setup/site-migration?siteSlug=foo.com` flow that includes actionable callbacks for the remaining events.
    * click on the `< Back` button in the flow and ensure the `calypso_signup_step_nav_back` event gets logged correctly
* from https://github.com/Automattic/wp-calypso/pull/93770, if we can ways that "go-to" or "go-next" or "exit-flow" are being used, then worth checking

## Pre-merge Checklist
